deeplake/util/path.py

```diff
@@ -103,6 +103,13 @@ def convert_pathlib_to_string_if_needed(path: Union[str, pathlib.Path]) -> str:
     return path
 
 
+def process_dataset_path(path: Union[str, pathlib.Path]) -> Tuple[str, Optional[str]]:
+    dataset_path, at, version = str(path).partition("@")
```

Comment:
we should disallow @ from dataset names if it isn't done already

Reply:
already done